src/router/router.js

```diff
@@ -135,7 +150,12 @@ export const navigate = async function () {
         holder.set('w', '100%')
         holder.set('h', '100%')
         // merge props with potential route params to be injected into the component instance
-        const props = { ...this[symbols.props], ...route.params, ...navigationData }
+        const props = {
+          ...this[symbols.props],
+          ...route.params,
+          ...navigationData,
+          ...beforeHookOutput,
```

Comment:
let's think carefully about the order here .. what should take precedence? `navigationData` or `beforeHookOutput`?

Reply:
I believe beforeHookOutput should take the highest priority because navigationData is provided at the time of declaring route definition, whereas beforeHookOutput can be the result from an API call.  

But I feel the best could be, users can still safely provide different keys in the beforeHookOutput object and navigationData object, ensuring that everything is available in options

Reply:
Yeah, I agree with your logic! :)